frontend/web/components/RolesTable.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { FC } from 'react'
+import CreateRole from './modals/CreateRole'
+import { useGetRolesQuery } from 'common/services/useRole'
+import { User, Role } from 'common/types/responses'
+import PanelSearch from './PanelSearch'
+import UserGroupStore from 'common/stores/user-group-store'
+import Button from './base/forms/Button'
+import ConfirmDeleteRole from './modals/ConfirmDeleteRole'
+import Icon from './Icon'
+import Panel from './base/grid/Panel'
+const rolesWidths = [250, 600, 100]
+
+type RolesTableType = {
+  organisationId: number
+  users: User[]
+}
+
+const RolesTable: FC<RolesTableType> = ({ organisationId, users }) => {
+  const groups = UserGroupStore.getGroups() // todo: this will become a hook
+  const { data: roles } = useGetRolesQuery(
+    { organisation_id: organisationId },
+    { skip: !organisationId },
+  )
+
+  const createRole = () => {
+    openModal(
+      'Create Role',
+      <CreateRole
+        organisationId={organisationId}
+        onComplete={() => {
+          toast('Role Created')
+          closeModal()
+        }}
+      />,
+      'side-modal',
+    )
+  }
+  const deleteRole = (role: Role) => {
+    openModal(
+      'Remove Role',
+      <ConfirmDeleteRole
+        role={role}
+        onComplete={() => {
+          toast('Role Deleted')
+        }}
+      />,
+      'p-0',
+    )
+  }
+
+  const editRole = (role: Role) => {
+    openModal(
+      'Edit Role and Permissions',
+      <CreateRole
+        organisationId={role.organisation}
+        isEdit
+        role={role}
+        onComplete={() => {
+          toast('Role Created')
+        }}
+        users={users}
+        groups={groups}
+      />,
+      'side-modal',
+    )
+  }
+  return (
+    <>
+      <Row space className='mt-4'>
+        <h5 className='m-b-0'>Roles</h5>
+        <Button
+          className='mr-2'
+          id='btn-invite'
+          onClick={() => createRole()}
+          type='button'
+        >
+          Create Role
+        </Button>
+      </Row>
+      <p className='fs-small lh-sm'>
+        Create custom roles, assign permissions, and keys to the role, and then
+        you can assign roles to users and/or groups.
```

Comment:
The comma here is hurting more than it's helping.
```suggestion
        Create custom roles, assign permissions and keys to the role, and then
        you can assign roles to users and/or groups.
```